server.js

```diff
@@ -1,2 +1,86 @@
-require('dotenv').config()
-const express = require('express')
\ No newline at end of file
+//Imports
+const mongoose = require("mongoose");
+const express = require("express");
+const path = require("path");
+const { MongoClient, ObjectId } = require("mongodb");
+require("dotenv").config();
+
+//Global variables
+const port = process.env.PORT || 5000;
+const staticDir = path.resolve("./client/public");
+
+//Server set-up
+const app = express();
+app.use(express.static(staticDir));
+app.use(express.urlencoded({ extended: true }));
+
+//Database Set-up
+mongoose.connect("mongodb://localhost:27017/log");
+//I guess we don't need these?:{ useNewUrlParser: true, useUnifiedTopology: true}
+
+const entrySchema = new mongoose.Schema({
+  title: String,
+  content: String,
+  tag: Array,
+  date: Date,
+});
+const EntriesModel = mongoose.model("entries", entrySchema);
+const entriesDB = mongoose.connection;
+
+//Port setup
+app.listen(port, () => {
+  console.log("listening on port", port);
+});
+
+//Post a new TIL entry
+app.post("/post", (request, response) => {
+  //create new object with request params, date
+  let newObj = {
+    date: Date.now(),
+    title: request.body.title,
+    content: request.body.content,
+    tag: request.body.tag,
+  };
+  //create new entry
+  const newEntry = new EntriesModel(newObj);
+  //save entry
+  newEntry.save(function (err) {
+    if (err) throw err;
+  });
+
+  //send 200 status and redirect
+  response.status(200).redirect(path.resolve("/"));
```

Comment:
Commenting this line out stops your page from breaking when the user submits their post. It leaves your page trying to reload forever but with this line here I get a "bad redirect" error page.